src/main/java/com/openlattice/chronicle/ChronicleStudyApi.java

```diff
@@ -75,32 +99,53 @@ UUID enrollSource(
      * @return Whether or not the datasource is tied to this participant in this study
      */
     //  we are leaving the study path in for now, because we don't know that participant's across studies are unique
+    @Deprecated( since = "apps v2" )
     @GET( BASE + STUDY_ID_PATH + PARTICIPANT_ID_PATH + DATASOURCE_ID_PATH )
     Boolean isKnownDatasource(
             @Path( STUDY_ID ) UUID studyId,
             @Path( PARTICIPANT_ID ) String participantId,
             @Path( DATASOURCE_ID ) String datasourceId );
 
+    /**
+     * Apps v2: Verifies that a participant in a study is associated with a specific data source.
+     *
+     * @param organizationId - Id of the organization to which study belongs
+     * @param studyId        - the study id
+     * @param participantId  - the participant id
+     * @param datasourceId   - the datasource (device) id
+     * @return Whether or not the datasource is tied to this participant in this study
+     */
+    @GET( BASE + ORGANIZATION_ID_PATH + STUDY_ID_PATH + PARTICIPANT_ID_PATH + DATASOURCE_ID_PATH + VALID )
+    Boolean isKnownDataSourceV2(
+            @Path( ORGANIZATION_ID ) UUID organizationId,
+            @Path( STUDY_ID ) UUID studyId,
+            @Path( PARTICIPANT_ID ) String participantId,
+            @Path( DATASOURCE_ID ) String datasourceId );
+
     /**
      * Verify that a participants is part of this study.
      *
-     * @param studyId       - the study id
-     * @param participantId - the participant id
+     * @param organizationId - Id of the organization to which study belongs
+     * @param studyId        - the study id
+     * @param participantId  - the participant id
      * @return Whether or not this participant is in this study
      */
-    @GET( BASE + STUDY_ID_PATH + PARTICIPANT_ID_PATH )
+    @GET( BASE + ORGANIZATION_ID_PATH + STUDY_ID_PATH + PARTICIPANT_ID_PATH + VALID )
```

Comment:
How come this method doesn't get a `V2` version? Actually, where is this being used?

Reply:
At the moment it's not used by any client (web/android)

Reply:
Sooo... let's remove it?

Reply:
I think we should keep it. I have been using it **_heavily_** for sanity check.

Reply:
Removed